README.md

```diff
@@ -42,87 +42,76 @@ Find a list of all the available options for your DogStatsD Client in the [DogSt
 ### Migrating from v4.x to v5.x
 
 If you are already using DogStatsD-ruby v4.x and you want to migrate to a version v5.x, the major
-change concerning you is the new threading model (please see section Threading model):
+change concerning you is the new [threading model](#threading-model):
 
 In practice, it means two things:
 
-1. Now that the client is buffering metrics before sending them, you have to manually
-call the method `Datadog::Statsd#flush` if you want to force the sending of metrics. Note that the companion thread will automatically flush the buffered metrics if the buffer gets full or when you are closing the instance.
+1. Now that the client is buffering metrics before sending them, you have to call `Datadog::Statsd#flush(sync: true)` if you want synchronous behavior. In most cases, this is not needed, as the companion thread will automatically flush the buffered metrics if the buffer gets full or when you are closing the instance.
 
 2. You have to make sure you are either:
 
-  * using singletons instances of the DogStatsD client and not allocating one each time you need one, letting the buffering mechanism flush metrics, it's still a bad solution if the process later forks (see related section below). Or,
-  * properly closing your DogStatsD client instance when it is not needed anymore using the method `Datadog::Statsd#close` to release the resources used by the instance and to close the socket
+  * Using a singleton instance of the DogStatsD client instead of creating a new instance whenever you need one; this will let the buffering mechanism flush metrics regularly
+  * Or properly disposing of the DogStatsD client instance when it is not needed anymore using the method `Datadog::Statsd#close`
 
 If you have issues with the companion thread or the buffering mode, you can instantiate a client that behaves exactly as in v4.x (i.e. no companion thread and flush on every metric submission):
 
 ```ruby
-# Import the library
-require 'datadog/statsd'
```

Comment:
This is still good for this simple example as you can copy/paste it into irb

Reply:
I don't mind re-adding it but... don't you think it distracts a bit from the point of the example? 🤔  

This being an example for `single_thread: true, buffer_max_pool_size: 1`, it seems really verbose -- I'm guessing if you got this far, you probably already know the basics, since this is advanced user stuff.

Reply:
If this was an example that was partial in lifecycle (e.g. only showing metrics sending) and/or not showing initialization/instantiation of the class, I would agree with you on the superfluous argument but it does seem like we're depicting the full scope (other than the metrics sending) in here, making it slightly more leaning for me towards having that line in there. I don't feel too strongly about it so it's fine either way.